pkg/agents/team7/agent1/agent.go

```diff
@@ -79,99 +96,495 @@ func New(baseAgent *infra.Base) (infra.Agent, error) {
+		Eaten, err := a.TakeFood(foodtotake)
+
+		if err != nil {
+			switch err.(type) {
+			case *infra.FloorError:
+			case *infra.NegFoodError:
+			case *infra.AlreadyEatenError:
+			default:
 			}
-			a.Log("Team 7 has seen the platform:", infra.Fields{"foodEaten": a.foodEaten, "health": a.HP(), "daysHungry": a.daysHungry})
-			a.seenPlatform = true
 		}
 
-		if (a.CurrPlatFood() == -1 && a.seenPlatform) || a.CurrPlatFood() == 100 {
-			//Get ready for new day
-			a.seenPlatform = false
-			a.prevHP = a.HP()
+		a.opMem.foodEaten = Eaten
+		if a.opMem.foodEaten > 0 {
+			a.behaviour.greediness -= int(0.5 * float64(a.opMem.daysHungry*a.opMem.daysHungry))
+			a.opMem.daysHungry = 0
+		} else {
+			a.behaviour.greediness -= int(0.3 * float64(a.opMem.daysHungry*a.opMem.daysHungry))
+			a.opMem.daysHungry++
+			a.behaviour.greediness += int(0.3 * float64(a.opMem.daysHungry*a.opMem.daysHungry))
 		}
 
+		a.Log("Team 7 has seen the platform:", infra.Fields{"foodEaten": a.opMem.foodEaten, "health": a.HP(), "daysHungry": a.opMem.daysHungry})
+		a.opMem.seenPlatform = true
+	}
+
+	if !a.PlatformOnFloor() && a.opMem.seenPlatform {
+		// reset variables Daily / prep for new day
+		a.opMem.seenPlatform = false
+		a.opMem.prevHP = a.HP()
+		a.opMem.foodEaten = 0
+		a.opMem.recievedReq = false
+		//a.opMem.messagesSent = false
 	}
 
-	//send Message
+	//End of Run()
+}
+
+// Message Handlers
+
+// Handle Asks
+func (a *CustomAgent7) HandleAskHP(msg messages.AskHPMessage) {
+	a.Log("Recieved askHP message from ", infra.Fields{"floor": msg.SenderFloor()})
+	reply := msg.Reply(a.ID(), a.Floor(), msg.SenderFloor(), a.HP())
+	a.SendMessage(reply)
+}
+
+func (a *CustomAgent7) HandleAskFoodTaken(msg messages.AskFoodTakenMessage) {
+	a.Log("Recieved askFoodTaken message from ", infra.Fields{"floor": msg.SenderFloor()})
+	reply := msg.Reply(a.ID(), a.Floor(), msg.SenderFloor(), int(a.opMem.foodEaten))
+	a.SendMessage(reply)
+}
+
+func (a *CustomAgent7) HandleAskIntendedFoodTaken(msg messages.AskIntendedFoodIntakeMessage) {
+	a.Log("Recieved askIntendedFoodTaken message from ", infra.Fields{"floor": msg.SenderFloor()})
+	reply := msg.Reply(a.ID(), a.Floor(), msg.SenderFloor(), int(health.FoodRequired(a.HP(), 60, a.HealthInfo())))
+	a.SendMessage(reply)
+}
+
+// Statements
+func (a *CustomAgent7) HandleStateFoodTaken(msg messages.StateFoodTakenMessage) {
+	statement := msg.Statement()
+	a.Log("Recieved a StateFoodTaken message from ", infra.Fields{"floor": msg.SenderFloor(), "statement": statement})
+}
+
+func (a *CustomAgent7) HandleStateHP(msg messages.StateHPMessage) {
+	statement := msg.Statement()
+	a.Log("I recieved a StateHP message from ", infra.Fields{"floor": msg.SenderFloor(), "statement": statement})
+}
+
+func (a *CustomAgent7) HandleStateIntendedFoodTaken(msg messages.StateIntendedFoodIntakeMessage) {
+	statement := msg.Statement()
+	a.Log("I recieved a StateIntendedFoodTaken message from ", infra.Fields{"floor": msg.SenderFloor(), "statement": statement})
+}
```

Comment:
If this is the only thing you're doing with State messages, I would rather you just not implement them so we save on computation